### PR TITLE
Feature/initial implementation

### PR DESCRIPTION
**Pull request by eheimbuch**
## Proposed changes
Create trace monitor and configuration

### Your checklist for this pull request

- [x] PR is well described
- [x] Related issues linked to PR if existing and labels set
- [x] Target branch is not master (in most cases develop should bet the target of choice) 
- [x] Code does not conflict with target branch
- [x] New code is covered with unit tests
- [x] CHANGELOG.md updated
- [x] Definition of Done's fulfilled: [DoD](https://github.com/scm-manager/scm-manager/blob/develop/docs/en/development/definition-of-done.md) // [UI DoD](https://github.com/scm-manager/scm-manager/blob/develop/docs/en/development/ui-dod.md)
- [x] Documentation updated (only necessary for new features or changed behaviour)

### Checklist for branch merge request (not required for forks)

- [ ] Branch is green/blue on [Jenkins](https://oss.cloudogu.com/jenkins/)
- [ ] Quality Gate passed on [SonarQube](https://sonarcloud.io/organizations/scm-manager/projects)

### CODE REVIEW

File: src/main/js/table.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+/*
```
Comment (sdorra) [inline src/main/js/table.ts:1]: Why is this file called table? I think this logic needs some unit tests.

**Review by sdorra — CHANGES_REQUESTED** [review]

File: src/main/java/com/cloudogu/scm/tracemonitor/TraceStore.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  public Collection<SpanContext> getAll() {
+    SecurityUtils.getSubject().checkPermission("traceMonitor:read");
+    DataStore<StoreEntry> store = createStore();
+    List<SpanContext> spans = new ArrayList<>();
+    store.getAll().values().forEach(entry -> spans.addAll(entry.getSpans()));
+    return spans;
+  }
+
+  public Collection<SpanContext> get(String kind) {
+    return getAll().stream()
```
[the remark this answers is not in the document]
Comment (eheimbuch) [reply src/main/java/com/cloudogu/scm/tracemonitor/TraceStore.java:65]: The check already happens in getAll()

File: src/main/js/SpanDetailsModal.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+type Props = {
+  active: boolean;
+  onClose: () => void;
+  modalData?: Span;
+};
+
+const SpanDetailsModal: FC<Props> = ({ onClose, modalData, active }) => {
+  const [t] = useTranslation("plugins");
+  const body = modalData ? (
+    <>
```
[the remark this answers is not in the document]
Comment (eheimbuch) [reply src/main/js/SpanDetailsModal.tsx:39]: Looks like i missed this on cleanup

File: src/main/js/TraceMonitor.tsx [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+import { apiClient, ErrorNotification, Loading, Title, Subtitle } from "@scm-manager/ui-components";
+import TraceMonitorTable from "./TraceMonitorTable";
+import { useTranslation } from "react-i18next";
+
+export type Span = {
+  kind: string;
+  opened: Date;
+  closed: Date;
+  durationInMillis: number;
+  labels: {};
```
[the remark this answers is not in the document]
Comment (eheimbuch) [reply src/main/js/TraceMonitor.tsx:34]: yeah, a type would be helpful

File: src/main/js/table.ts (hunk 1 above)
Comment (eheimbuch) [reply src/main/js/table.ts:1]: Naming.. It was time.ts at first and after some refactorings i kinda stuck at table. Suggestion?

Comment (sdorra) [reply src/main/js/table.ts:1]: `time.ts` sounds good to me

**Review by sdorra — APPROVED** [review]

**Pull request merged**